### PR TITLE
Execute Procedure Expect Exception in standalone mode

### PR DESCRIPTION
**Pull request by MMatten**
Adds support for the `Execute Procedure Expect Exception` in standalone mode.

### CODE REVIEW

File: FitNesseRoot/DbFit/AcceptanceTests/JavaTests/OracleTests/LibraryTests/StoredProcWithException/content.txt [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+!3 Use ''Execute Procedure Expect Exception'' to check if a stored proc will throw an exception
 
 !|Execute|!-
 create or replace procedure TestProc1(name varchar2, strlength out number) as
 begin
-        if (name='xx') then
-          raise_application_error(-20001,'test exception');
-        end if;
-	strlength:=length(name);
+    if (name='xx') then
```
Comment (javornikolov) [inline FitNesseRoot/DbFit/AcceptanceTests/JavaTests/OracleTests/LibraryTests/StoredProcWithException/content.txt]: If we're touching this, might be good to refine a few other things: add space around `=`, `:=`, and after `,`.

File: dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    public ExecuteProcedureExpectException(DBEnvironment dbEnvironment, String procName) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = false;
+    }
+
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        if (!excNumberDefined) return ExpectedBehaviour.ANY_EXCEPTION;
+        return ExpectedBehaviour.SPECIFIC_EXCEPTION;
```
Comment (javornikolov) [inline dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java]: May we simplify that a bit? It should be possible to fit it in a single 1-line operator.

File: FitNesseRoot/DbFit/AcceptanceTests/JavaTests/OracleTests/LibraryTests/StoredProcWithException/content.txt (hunk 1 above)
Comment (MMatten) [inline FitNesseRoot/DbFit/AcceptanceTests/JavaTests/OracleTests/LibraryTests/StoredProcWithException/content.txt]: Now applied.

Perhaps we should adopt some standard SQL formatting via some free web service such as http://www.dpriver.com/pp/sqlformat.htm, but I don't know how we could define a standard config of our chosing. Perhaps use the service's default. Or maybe there are other better free desktop tools.

File: dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java (hunk 2 above)
Comment (MMatten) [inline dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java]: Updated.

**Comment by MMatten** [comment]
I've added some missing content/properties (now .wiki) files too for the new `LibraryTests` root and fixed a couple of issues with the symlinks.

File: dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    public ExecuteProcedureExpectException(DBEnvironment dbEnvironment, String procName) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = false;
+    }
+
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        return excNumberDefined ? ExpectedBehaviour.SPECIFIC_EXCEPTION : ExpectedBehaviour.ANY_EXCEPTION;
```
Comment (javornikolov) [inline dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java]: Maybe would look better if we can skip the prefix `ExpectedBehaviour.` here?

**Comment by javornikolov** [comment]
> I've added some missing content/properties (now .wiki) files too for the new LibraryTests root and fixed a couple of issues with the symlinks.

What do you think about extracting this refactoring and small fixes to a separate PR? This would keep the current PR focused on the Execute Procedure Expect Exception change.

**Comment by MMatten** [comment]
These changes are directly related to the changes to the tests (creating the `LibraryTests` container). They weren't existing issues.

Sent from my iPhone

On 18 Mar 2017, at 13:15, Yavor Nikolov <notifications@github.com<mailto:notifications@github.com>> wrote:


I've added some missing content/properties (now .wiki) files too for the new LibraryTests root and fixed a couple of issues with the symlinks.

What do you think about extracting this refactoring and small fixes to a separate PR? This would keep the current PR focused on the Execute Procedure Expect Exception change.

**Comment by javornikolov** [comment]
> These changes are directly related to the changes to the tests (creating the `LibraryTests` container). They weren't existing issues.

Ah, OK then.

File: dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java (hunk 3 above)
Comment (MMatten) [inline dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java]: Agreed. What do you think of it now?

Is it better to only import the minimal number of references?

Comment (javornikolov) [inline dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java]: Yes, it looks good now. What do you mean by min number of references? (In general I think we have a Checkstyle will raise an error if we have unused imports).

**Comment by MMatten** [comment]
I meant the use of import with `.*` instead of specific items.


On 18 Mar 2017, at 13:46, Yavor Nikolov <notifications@github.com<mailto:notifications@github.com>> wrote:


@javornikolov commented on this pull request.

________________________________

In dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java<https://github.com/dbfit/dbfit/pull/556#discussion_r106781225>:

> +                            int expectedErrorCode) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = true;
+        this.excNumberExpected = expectedErrorCode;
+    }
+
+    public ExecuteProcedureExpectException(DBEnvironment dbEnvironment, String procName) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = false;
+    }
+
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        return excNumberDefined ? ExpectedBehaviour.SPECIFIC_EXCEPTION : ExpectedBehaviour.ANY_EXCEPTION;


Yes, it looks good now. What do you mean by min number of references? (In general I think we have a Checkstyle will raise an error if we have unused imports).

**Comment by javornikolov** [comment]
> I meant the use of import with `.*` instead of specific items.

It depends. In general it's not very good to "pollute" the namespace with too many things (there might be naming conflicts). In case of enums in particular (which is the case here) - I guess it would be usually OK to do a static * import.

**Comment by javornikolov** [comment]
The current state seems OK for this PR to be merged after some squashing of the fix-up commits.

**Comment by MMatten** [comment]
> The current state seems OK for this PR to be merged after some squashing of the fix-up commits.

Ooops. Squashed.

**Comment by javornikolov** [comment]
All looks great now :+1: Merging... Thank you @MMatten!

**Pull request merged**